### PR TITLE
[IMP] Recruitment: updating job positions doc for 16 version

### PR DESCRIPTION
**Pull request by larm-odoo**
Lots changed especially to the interview form instructions.

### CODE REVIEW

File: content/applications/hr/recruitment/new_job.rst [hunk 1]
```diff
@@ -143,68 +173,274 @@ Sections are indicated by a gray background, while questions have a white backgr
...
+  Message` and :guilabel:`Comment is an answer` field appear if activated. Enter the text to appear
+  in the :guilabel:`Comment Message` field. This should be helpful to the applicant, such as `If
+  other, please specify`. Last, if the comment should be logged as the applicant's answer, activate
+  the box.
+- :guilabel:`Multiple Lines Text Box`: A :guilabel:`Placeholder` field appears. Enter any text that
+  should appear with the question to help clarify how the applicant should answer.
+- :guilabel:`Single Line Text Box` and :guilabel:`Numerical Value`: A :guilabel:`Validate entry`
+  option appears. Activate this option if what the applicant enters must be a numerical value, *and*
+  needs to be verified. If the answer is *not* a numerical value, the answer cannot be verified (do
+  not activate the :guilabel:`Validate entry` option). When activated, several other options appear.
```
Comment (jero-odoo) [inline content/applications/hr/recruitment/new_job.rst:280]: Double negative... I would rephrase as "can **only** be verified if answer is numerical" or something similar

File: content/applications/hr/recruitment/new_job.rst [hunk 2]
```diff
@@ -143,68 +173,274 @@ Sections are indicated by a gray background, while questions have a white backgr
...
+  - :guilabel:`Randomized per Section`: Display only a random selection of questions from each
+    section.
+
+
+- :guilabel:`Back Button`: Activate this option if the applicant is able to click a back button to
+  go back to previous questions.
+
+.. note::
+   Although it is a selectable option, the :guilabel:`Randomized per section` option is not
+   typically selected for an interview form.
```
Comment (jero-odoo) [inline content/applications/hr/recruitment/new_job.rst:380]: I mentioned this on the review for 15, but instead of saying "not typically selected" I would emphasize that this shouldn't be selected unless they are ok with receiving surveys that are incomplete, as it means not all of the included questions will be presented to applicants. (Instead of just saying don't do it, say why)

**Review by jero-odoo — CHANGES_REQUESTED** [review]
hey @larm-odoo similar comments as on v15, mostly formatting. Let me know if you have any questions. Thanks!

File: content/applications/hr/recruitment/new_job.rst (hunk 2 above)
Comment (larm-odoo) [reply content/applications/hr/recruitment/new_job.rst:380]: This is npw changed to how I changed it in 15

**Review by StraubCreative — APPROVED** [review]
@robodoo r+
